src/VirtualClient/VirtualClient.Actions/MLPerfTraining/MLPerfTrainingExecutor.cs

```diff
@@ -0,0 +1,517 @@
+            if (string.IsNullOrEmpty(this.DiskFilter))
+            {
+                this.DiskFilter = "SizeGreaterThan:1000gb&OSDisk:false";
+            }
+        }
+
+        /// <summary>
+        /// Disk filter string to filter disks to format.
+        /// </summary>
+        public string DiskFilter
+        {
+            get
+            {
+                // Change the size -----------------------------------------------------------------------------------------------------------------------------
+                string filter = this.Parameters.GetValue<string>(nameof(MLPerfTrainingExecutor.DiskFilter), "SizeGreaterThan:1000gb");
+                // Enforce filter to remove OS disk.
+                filter = $"{filter}&OSDisk:false";
+                return filter;
+            }
+
+            set
+            {
+                this.Parameters[nameof(MLPerfTrainingExecutor.DiskFilter)] = value;
+            }
+        }
+
+        /// <summary>
+        /// The MLPerf Training model name (e.g. bert, rnnt, ssd-mobilenet).
+        /// </summary>
+        public string Model
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.Model));
+            }
+        }
+
+        /// <summary>
+        /// The current running user
+        /// </summary>
+        public string Username
+        {
+            get
+            {
+                string username = this.Parameters.GetValue<string>(nameof(MLPerfTrainingExecutor.Username));
+                if (string.IsNullOrWhiteSpace(username))
+                {
+                    username = this.GetCurrentUserName(true);
+                }
+
+                return username;
+            }
+        }
+
+        /// <summary>
+        /// ML specific parameter used to control size of training samples
+        /// </summary>
+        public string BatchSize
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.BatchSize));
+            }
+        }
+
+        /// <summary>
+        /// The current running user
+        /// </summary>
+        public string Implementation
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.Implementation));
+            }
+        }
+
+        /// <summary>
+        /// Container image name
+        /// </summary>
+        public string ContainerName
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.ContainerName));
+            }
+        }
+
+        /// <summary>
+        /// Container image name
+        /// </summary>
+        public string DataPath
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.DataPath));
+            }
+        }
+
+        /// <summary>
+        /// Number of GPUs to be utilized
+        /// </summary>
+        public string GPUNum
```

Comment:
Lets give a better name to it, GPUNum is confusing with number of GPUs or that index of GPU, what do you think of NumberOfGPUs.

Another question, let say we have a machine with 2 A100 and 2 A10 GPUs, if i give this value as 2, what will happen in that case?

Reply:
I have updated the name

In the case where we have 2 GPUs of 2 different kinds, it will take the first 2 GPUs, numbered 0 and 1 when it lists them.